zfs/zfs.go

```diff
@@ -814,14 +844,28 @@ func ZFSSend(ctx context.Context, sendArgs ZFSSendArgs) (*ReadCloserCopier, erro
 	args = append(args, "send")
 
 	// pre-validation of sendArgs for plain ErrEncryptedSendNotSupported error
+	// we tie BackupProperties (send -b) and SendRaw (-w, same as with Encrypted) to this
+	// since these were released together.
 	// TODO go1.13: push this down to sendArgs.Validate
-	if encryptedSendValid := sendArgs.Encrypted.Validate(); encryptedSendValid == nil && sendArgs.Encrypted.B {
-		supported, err := EncryptionCLISupported(ctx)
+	if sendArgs.Encrypted.IsTrue() || sendArgs.Raw.IsTrue() || sendArgs.BackupProperties.IsTrue() {
+		encryptionSupported, err := EncryptionCLISupported(ctx)
 		if err != nil {
 			return nil, errors.Wrap(err, "cannot determine CLI native encryption support")
 		}
-		if !supported {
-			return nil, ErrEncryptedSendNotSupported
+
+		if !encryptionSupported {
+			if sendArgs.BackupProperties.IsTrue() || sendArgs.Encrypted.IsTrue() {
+				return nil, ErrEncryptedSendNotSupported
+			}
+
+			if sendArgs.Raw.IsTrue() {
+				// zfs-send(8) about `send -w`: "For unencrypted datasets, this flag will be equivalent to -Lec."
+				// TODO: expose via config?
+				// TODO: definitely add feature checks (also for -b!)
+				sendArgs.LargeBlocks = &NilBool{true}
+				sendArgs.EmbeddedData = &NilBool{true}
+				sendArgs.Compressed = &NilBool{true}
```

Comment:
`sendArgs` should not be mutated in `ZFSSend`: it expresses intent of the caller. I can't seem to find the place where you convert the `config.SendOptions.Raw` to a `ZFSSendArgs.Raw`. Wherever that is / will be, that's where above smartness should be. Plus some unit tests for that smartness ;)